packages/mana/weightedbasevector.go

```diff
@@ -0,0 +1,317 @@
+package mana
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/iotaledger/hive.go/identity"
+	"golang.org/x/xerrors"
+)
+
+// WeightedBaseManaVector represents a base mana vector
+type WeightedBaseManaVector struct {
+	vector map[identity.ID]*WeightedBaseMana
+	weight float64
+	target Type
+	sync.RWMutex
+}
+
+// Type returns the type of this mana vector.
+func (w *WeightedBaseManaVector) Type() Type {
+	return WeightedMana
+}
+
+// Target returns the target type of this mana vector, namely if it takes into account access or consensus pledges.
+func (w *WeightedBaseManaVector) Target() Type {
+	return w.target
+}
+
+// Size returns the size of this mana vector.
+func (w *WeightedBaseManaVector) Size() int {
+	w.RLock()
+	defer w.RUnlock()
+	return len(w.vector)
+}
+
+// SetWeight sets the weight for the while vector.
+func (w *WeightedBaseManaVector) SetWeight(weight float64) error {
+	if weight < 0.0 || weight > 1.0 {
```

Comment:
Can these min and max weights be constants?

Reply:
Yes, they can.